rust/cubesql/cubesql/src/compile/rewrite/rules/split.rs

```diff
@@ -1879,4 +1981,65 @@ impl SplitRules {
             false
         }
     }
+
+    fn transform_aggr_fun_with_literal(
+        &self,
+        fun_expr_var: &'static str,
+        expr_val: &'static str,
+    ) -> impl Fn(&mut EGraph<LogicalPlanLanguage, LogicalPlanAnalysis>, &mut Subst) -> bool {
+        let fun_expr_var = var!(fun_expr_var);
+        let expr_val = var!(expr_val);
+        move |egraph, subst| {
+            for fun in var_iter!(egraph[subst[fun_expr_var]], AggregateFunctionExprFun) {
+                match fun {
+                    AggregateFunction::Count | AggregateFunction::Sum => (),
+                    _ => return true,
+                };
+
+                for expr in var_iter!(egraph[subst[expr_val]], LiteralExprValue) {
+                    match expr {
+                        ScalarValue::Utf8(None) | ScalarValue::LargeUtf8(None) => return true,
+                        ScalarValue::Utf8(value) | ScalarValue::LargeUtf8(value) => match value {
```

Comment:
I have concerns regarding this match, but let's go with this solution for the time being.